pkg/store/gcs.go

```diff
@@ -385,40 +354,71 @@ func (s *GCSStore) Series(ctx context.Context, req *storepb.SeriesRequest) (*sto
 			indexr  = b.indexReader()
 			chunkr  = b.chunkReader(ctx)
 		)
+		defer indexr.Close()
+		defer chunkr.Close()
 		// TODO(fabxc): only aggregate chunk metas first and add a subsequent fetch stage
 		// where we consolidate requests.
 		g.Go(func() error {
-			defer indexr.Close()
-			defer chunkr.Close()
+			var preloadc int
 
 			set, err := tsdb.LookupChunkSeries(indexr, nil, matchers...)
 			if err != nil {
 				return errors.Wrap(err, "get series set")
 			}
-			var chks []tsdb.ChunkMeta
+			resSet := &gcsSeriesSet{chunkr: chunkr}
```

Comment:
Minor readability issue: 
Can we create here just empty `set    []seriesEntry`? we totally don't need whole `gcsSeriesSet` here, we can create it in `resSet.set = append(resSet.set, s) line` .. Because right now I initially thought you forgot to add `resSet.set`, but actually you are adding that in loop below. Looks like we don't need whole `gcsSeriesSet` yet